woocommerce/woocommerce-ios
# Grouped products are now editable when Products M3 flag is enabled

Grouped product form for #2199 

## Changes

- Added a new icon `UIImage.widgetsImage`, with asset `icon-widgets` exported from the Master Component
- Implemented actions for a grouped product in `ProductFormActionsFactory` + test case
- Implemented view model for grouped products row
- Showed product form for a grouped product when M3 flag is enabled in `ProductDetailsFactory` + test cases

## Testing

- Go to the Products tab
- Tap on a grouped product --> the grouped products row should always be visible, with optional rows (SKU, brief description, categories) that could also be in the bottom sheet
- Try editing the grouped product - name, image, SKU, etc.
- Tap "Update" --> the grouped product should be updated remotely

## Example screenshots

Note: action for editing grouped products will be implemented separately as a subtask of #2199

mode | > 1 product | 1 product | no linked products
-- | -- | -- | --
light | ![Simulator Screen Shot - iPhone 11 - 2020-06-19 at 08 44 01](https://user-images.githubusercontent.com/1945542/85085839-a02e6180-b20b-11ea-903f-a23f0f0ab242.png) | ![Simulator Screen Shot - iPhone 11 - 2020-06-19 at 08 44 21](https://user-images.githubusercontent.com/1945542/85085845-a3c1e880-b20b-11ea-97c2-288778419b23.png) | ![Simulator Screen Shot - iPhone 11 - 2020-06-19 at 08 44 31](https://user-images.githubusercontent.com/1945542/85085846-a45a7f00-b20b-11ea-9f41-6809de923be4.png)
dark | ![Simulator Screen Shot - iPhone 11 - 2020-06-19 at 08 44 46](https://user-images.githubusercontent.com/1945542/85085849-a4f31580-b20b-11ea-8148-3479c1a4bfeb.png) | ![Simulator Screen Shot - iPhone 11 - 2020-06-19 at 08 44 51](https://user-images.githubusercontent.com/1945542/85085851-a58bac00-b20b-11ea-84cc-7f8e8dda7a9d.png) | ![Simulator Screen Shot - iPhone 11 - 2020-06-19 at 08 44 56](https://user-images.githubusercontent.com/1945542/85085852-a6244280-b20b-11ea-86b8-268f8d2d61a9.png)
bottom sheet | ![Simulator Screen Shot - iPhone 11 - 2020-06-19 at 09 00 19](https://user-images.githubusercontent.com/1945542/85085854-a6bcd900-b20b-11ea-95d1-098009c012b2.png) | ![Simulator Screen Shot - iPhone 11 - 2020-06-19 at 09 00 29](https://user-images.githubusercontent.com/1945542/85085855-a7556f80-b20b-11ea-9ca2-e7cd73fe3c76.png) | ![Simulator Screen Shot - iPhone 11 - 2020-06-19 at 09 00 37](https://user-images.githubusercontent.com/1945542/85085857-a7ee0600-b20b-11ea-947d-cda00ffd05c4.png)




Update release notes:
